### PR TITLE
cleanup C++0x support in irrUString.h

### PR DESCRIPTION
**Pull request by nerzhul**

### CODE REVIEW

File: include/irrUString.h [hunk 1]
```diff
@@ -83,17 +72,10 @@ static const irr::u16 UTF16_LO_SURROGATE = 0xDC00;
 
 namespace irr
 {
-
 	// Define our character types.
-#ifdef USTRING_CPP0X_NEWLITERALS	// C++0x
```
Comment (sfan5) [inline include/irrUString.h:88]: other way around, `char32_t` is a C++11 type so we should use it

Comment (nerzhul) [reply include/irrUString.h:88]: @sfan5 you mean we should keep the mapping here ?

File: include/irrUString.h [hunk 2]
```diff
@@ -886,10 +853,7 @@ class ustring16
 		for (; first != last; ++first)
 			append((uchar32_t)*first);
 	}
-#endif
 
-
-#ifndef USTRING_CPP0X_NEWLITERALS
```
Comment (sfan5) [inline include/irrUString.h:892]: If you remove <i>exactly</i> the function inside `#ifndef USTRING_CPP0X_NEWLITERALS` then it should work

File: include/irrUString.h [hunk 3]
```diff
@@ -2091,28 +2019,6 @@ class ustring16
 	}
 
 
-#ifdef USTRING_CPP0X_NEWLITERALS
```
Comment (sfan5) [inline include/irrUString.h:2094]: There's also a few more blocks of code that should stay.

**Pull request merged**